src/main/java/org/terasology/tutorialWorldGeneration/TutorialWorldRasterizer.java

```diff
@@ -42,12 +42,12 @@ public void initialize() {
     public void generateChunk(CoreChunk chunk, Region chunkRegion) {
         ElevationFacet elevationFacet = chunkRegion.getFacet(ElevationFacet.class);
         SurfacesFacet surfacesFacet = chunkRegion.getFacet(SurfacesFacet.class);
-        for (Vector3i position : chunkRegion.getRegion()) {
-            float surfaceHeight = elevationFacet.getWorld(position.x, position.z);
-            if (surfacesFacet.getWorld(JomlUtil.from(position))) {
-                chunk.setBlock(ChunkMath.calcRelativeBlockPos(position), grass);
-            } else if (position.y < surfaceHeight) {
-                chunk.setBlock(ChunkMath.calcRelativeBlockPos(position), dirt);
+        for (Vector3ic position : chunkRegion.getRegion()) {
+            float surfaceHeight = elevationFacet.getWorld(position.x(), position.z());
+            if (surfacesFacet.getWorld(position)) {
+                chunk.setBlock(Chunks.toRelative(position), grass);
+            } else if (position.y() < surfaceHeight) {
+                chunk.setBlock(Chunks.toRelative(position), dirt);
```

Comment:
I think we need to provide Vector3ic to chunk.setBlock() method in lines 48 and 50. But Chunks.toRelative(position) returns Vector3i.

Reply:
`Vector3i` is a **subtype** of `Vector3ic` - you can use the non-constant `Vector3i` wherever the constant variant is required. :nerd_face: 

The other way this does not hold. In that case, we have to turn the constant vector into a mutable one - by allocating a new vector instance and making a copy of our vector like so:

```java
Vector3i mutableVector = new Vector3i(constantVector);
```

Reply:
A problem in this line could be that `position` is immutable but `Chunks.toRelative(...)` tries to modify it. There are multiple ways to solve this, for instance:

```java
// pass a mutable copy of 'position' to the method
Chunks.toRelative(new Vector3i(position))

// pass the immutable vector as is, but provide a new mutable vector to store the result in
Chunks.toRelative(position, new Vector3i())
```

In case this all happens in a loop we can make a small performance improvement by re-using the same mutable vector in some cases. But one step at a time :wink:

Reply:
Yes changing it to
 chunk.setBlock(Chunks.toRelative(new Vector3i(position)), dirt); 
fixed the problem for me.